// Copyright 2019 Xcalar, Inc. All rights reserved.
//
// No use, or distribution, of this source code is permitted in any form or
// means without a valid, written license agreement with Xcalar, Inc.
// Please refer to the included "COPYING" file for terms and conditions
// regarding the use and redistribution of this software.
//

syntax = "proto3";

import "google/protobuf/empty.proto";
import "xcalar/compute/localtypes/Workbook.proto";

package xcalar.compute.localtypes.DagNode;

message DagNodeInputMsg {
    string dag_node_name = 1;
    xcalar.compute.localtypes.Workbook.WorkbookScope scope = 2;
}

service DagNode {
    rpc Pin (DagNodeInputMsg) returns (google.protobuf.Empty);
    rpc Unpin (DagNodeInputMsg) returns (google.protobuf.Empty);
};
